packages/@n8n/nodes-langchain/nodes/llms/N8nLlmTracing.ts

```diff
@@ -120,47 +124,43 @@ export class N8nLlmTracing extends BaseCallbackHandler {
 		}
 
 		const parsedMessages =
-			typeof this.lastInput.messages === 'string'
-				? this.lastInput.messages
-				: this.lastInput.messages.map((message) => {
+			typeof runDetails.messages === 'string'
+				? runDetails.messages
+				: runDetails.messages.map((message) => {
 						if (typeof message === 'string') return message;
 						if (typeof message?.toJSON === 'function') return message.toJSON();
 
 						return message;
 					});
 
-		this.executionFunctions.addOutputData(this.connectionType, this.lastInput.index, [
+		this.executionFunctions.addOutputData(this.connectionType, runDetails.index, [
 			[{ json: { ...response } }],
 		]);
 		void logAiEvent(this.executionFunctions, 'n8n.ai.llm.generated', {
 			messages: parsedMessages,
-			options: this.lastInput.options,
+			options: runDetails.options,
 			response,
 		});
 	}
 
-	async handleLLMStart(llm: Serialized, prompts: string[]) {
+	async handleLLMStart(llm: Serialized, prompts: string[], runId: string) {
 		const estimatedTokens = await this.estimateTokensFromStringList(prompts);
 
 		const options = llm.type === 'constructor' ? llm.kwargs : llm;
-		const { index } = this.executionFunctions.addInputData(
-			this.connectionType,
+		const { index } = this.executionFunctions.addInputData(this.connectionType, [
 			[
-				[
-					{
-						json: {
-							messages: prompts,
-							estimatedTokens,
-							options,
-						},
+				{
+					json: {
+						messages: prompts,
+						estimatedTokens,
+						options,
 					},
-				],
+				},
 			],
-			this.lastInput.index + 1,
```

Comment:
Was this parameter not used?

Reply:
It didn't seem to have an effect and it's optional anyway - the method itself returns the assigned index which we'll use to populate output index